### PR TITLE
JDK-8301077: Replace NULL with nullptr in share/services/

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory share/services/. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301077](https://bugs.openjdk.org/browse/JDK-8301077): Replace NULL with nullptr in share/services/

### Reviewers
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12189/head:pull/12189` \
`$ git checkout pull/12189`

Update a local copy of the PR: \
`$ git checkout pull/12189` \
`$ git pull https://git.openjdk.org/jdk pull/12189/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12189`

View PR using the GUI difftool: \
`$ git pr show -t 12189`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12189.diff">https://git.openjdk.org/jdk/pull/12189.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/services/attachListener.cpp [hunk 1]
```diff
@@ -365,7 +365,7 @@ static AttachOperationFunctionInfo funcs[] = {
   { "setflag",          set_flag },
   { "printflag",        print_flag },
   { "jcmd",             jcmd },
-  { NULL,               NULL }
+  { nullptr,               nullptr }
```
Comment (jdksjolen) [inline src/hotspot/share/services/attachListener.cpp:368]: Align

File: src/hotspot/share/services/classLoadingService.cpp [hunk 2]
```diff
@@ -45,10 +45,10 @@
 #define HOTSPOT_CLASS_loaded HOTSPOT_CLASS_LOADED
 #define DTRACE_CLASSLOAD_PROBE(type, clss, shared)  \
   {                                                 \
-    char* data = NULL;                              \
+    char* data = nullptr;                              \
     int len = 0;                                    \
     Symbol* name = (clss)->name();                  \
-    if (name != NULL) {                             \
+    if (name != nullptr) {                             \
```
Comment (jdksjolen) [inline src/hotspot/share/services/classLoadingService.cpp:51]: Align

File: src/hotspot/share/services/heapDumper.cpp [hunk 3]
```diff
@@ -789,10 +789,10 @@ class ParDumpWriter : public AbstractDumpWriter {
 
  private:
   void allocate_internal_buffer() {
-    assert(_buffer_queue != NULL, "Internal buffer queue is not ready when allocate internal buffer");
-    assert(_buffer == NULL && _buffer_base == NULL, "current buffer must be NULL before allocate");
+    assert(_buffer_queue != nullptr, "Internal buffer queue is not ready when allocate internal buffer");
+    assert(_buffer == nullptr && _buffer_base == nullptr, "current buffer must be nullptr before allocate");
```
Comment (jdksjolen) [inline src/hotspot/share/services/heapDumper.cpp:793]: String

File: src/hotspot/share/services/heapDumper.cpp [hunk 4]
```diff
@@ -1290,10 +1290,10 @@ void DumperSupport::dump_instance_class(AbstractDumpWriter* writer, Klass* k) {
 
 // creates HPROF_GC_CLASS_DUMP record for the given array class
 void DumperSupport::dump_array_class(AbstractDumpWriter* writer, Klass* k) {
-  InstanceKlass* ik = NULL; // bottom class for object arrays, NULL for primitive type arrays
+  InstanceKlass* ik = nullptr; // bottom class for object arrays, null for primitive type arrays
```
Comment (jdksjolen) [inline src/hotspot/share/services/heapDumper.cpp:1293]: string

**Review by jdksjolen — COMMENTED** [review]
Found some things to manually fix.

**Comment by jdksjolen** [comment]
Passes tier1.

File: src/hotspot/share/services/heapDumper.cpp [hunk 5]
```diff
@@ -1847,7 +1847,7 @@ class VM_HeapDumper : public VM_GC_Operation, public WorkerTask {
   }
 
   void prepare_parallel_dump(uint num_total) {
-    assert (_dumper_controller == NULL, "dumper controller must be NULL");
+    assert (_dumper_controller == nullptr, "dumper controller must be nullptr");
```
Comment (plummercj) [inline src/hotspot/share/services/heapDumper.cpp:1850]: The string should say "null", not "nullptr".

File: src/hotspot/share/services/mallocTracker.hpp [hunk 6]
```diff
@@ -329,11 +329,11 @@ class MallocTracker : AllStatic {
   static bool print_pointer_information(const void* p, outputStream* st);
 
   static inline MallocHeader* malloc_header(void *memblock) {
-    assert(memblock != NULL, "NULL pointer");
+    assert(memblock != nullptr, "nullptr pointer");
```
Comment (plummercj) [inline src/hotspot/share/services/mallocTracker.hpp:332]: String should say "null", not "nullptr"

File: src/hotspot/share/services/mallocTracker.hpp [hunk 7]
```diff
@@ -329,11 +329,11 @@ class MallocTracker : AllStatic {
...
     return (MallocHeader*)((char*)memblock - sizeof(MallocHeader));
   }
   static inline const MallocHeader* malloc_header(const void *memblock) {
-    assert(memblock != NULL, "NULL pointer");
+    assert(memblock != nullptr, "nullptr pointer");
```
Comment (plummercj) [inline src/hotspot/share/services/mallocTracker.hpp:336]: String should say "null", not "nullptr"

File: src/hotspot/share/services/management.cpp [hunk 8]
```diff
@@ -1642,8 +1642,8 @@ class ThreadTimesClosure: public ThreadClosure {
 
 ThreadTimesClosure::ThreadTimesClosure(objArrayHandle names,
                                        typeArrayHandle times) {
-  assert(names() != NULL, "names was NULL");
-  assert(times() != NULL, "times was NULL");
+  assert(names() != nullptr, "names was nullptr");
+  assert(times() != nullptr, "times was nullptr");
```
Comment (plummercj) [inline src/hotspot/share/services/management.cpp:1646]: String should say "null", not "nullptr"

File: src/hotspot/share/services/management.cpp [hunk 9]
```diff
@@ -1657,7 +1657,7 @@ ThreadTimesClosure::ThreadTimesClosure(objArrayHandle names,
 //
 void ThreadTimesClosure::do_thread(Thread* thread) {
   assert(Threads_lock->owned_by_self(), "Must hold Threads_lock");
-  assert(thread != NULL, "thread was NULL");
+  assert(thread != nullptr, "thread was nullptr");
```
Comment (plummercj) [inline src/hotspot/share/services/management.cpp:1660]: String should say "null", not "nullptr"

File: src/hotspot/share/services/threadService.cpp [hunk 10]
```diff
@@ -221,7 +221,7 @@ void ThreadService::current_thread_exiting(JavaThread* jt, bool daemon) {
 
 // FIXME: JVMTI should call this function
 Handle ThreadService::get_current_contended_monitor(JavaThread* thread) {
-  assert(thread != NULL, "should be non-NULL");
+  assert(thread != nullptr, "should be non-nullptr");
```
Comment (plummercj) [inline src/hotspot/share/services/threadService.cpp:224]: String should say "null", not "nullptr"

**Review by plummercj — CHANGES_REQUESTED** [review]

**Comment by jdksjolen** [comment]
Thanks @plummercj , I've pushed your fixes.

**Review by plummercj — APPROVED** [review]

**Comment by jdksjolen** [comment]
Still passes tier1!

/integrate

**Pull request closed without merging**